### ISSUE TITLE
How the predictions can be related to input data?

### ISSUE THREAD

**Issue by vabhishekprdxn** [tensorflow/ranking#92]
Following the issue#69 I am able to get the predictions but I am not getting how it can be related to the input data as predictions keep on shuffling with in the list. I am using Jupyter notebook and I tried removing shuffle from the line **train_dataset = train_dataset.shuffle(1000).repeat().batch(_BATCH_SIZE_)** in input_fn but it didn't worked. Please help in getting prediction corresponding to document in the list.

**Comment by DianeHu** [comment]
I also had this problem. There's a shuffle hidden in libsvm_generator if you're using that. If you remove the np.random.shuffle(doc_list) in data.py in the function _libsvm_generate(num_features, list_size, doc_list), that will fix the shuffling and get you predictions corresponding to the input order.

**Comment by vabhishekprdxn** [comment]
Yeah got that! It's working now as expected. Thanks for the help.

**Comment by mayank953** [comment]
@DianeHu  Will the shuffle in libsvm_generator also pose a problem if I make predictions using saved_model_cli for multiple entries or if I load the model in python & then make the predictions?

**Comment by DianeHu** [comment]
@mayank953 I'm not sure, but if any of your data passes through the libsvm_generator at any point then I imagine it would be an issue regardless on prediction. For example, if you reload from a warm started checkpoint, the shuffle will still be an issue.
